src/components/Divider/Divider.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import styled, { StyledComponent } from 'styled-components';
+import { compose, space, SpaceProps } from 'styled-system';
+import { theme } from '../../essentials/theme';
+import { get } from '../../utils/themeGet';
+
+type DividerOffset = number | string;
+
+interface DividerProps extends SpaceProps {
+    /**
+     * Set the direction of the divider to vertical
+     */
+    vertical?: boolean;
+    /**
+     * Set offset / margin of the divider from the surrounding content
+     */
+    offset?: DividerOffset;
+}
+
+const HorizontalLine: StyledComponent<'div', typeof theme, DividerProps, 'theme'> = styled.div.attrs({ theme })<
+    Pick<SpaceProps, 'my'>
+>`
+    width: 100%;
+    margin-left: auto;
+    margin-right: auto;
+    border: 0;
+    border-top: 1px solid ${get('semanticColors.border.primary')};
+
+    ${compose(space)}
+`;
+
+const VerticalLine: StyledComponent<'div', typeof theme, DividerProps, 'theme'> = styled.div.attrs({ theme })<
+    Pick<SpaceProps, 'mx'>
+>`
+    display: inline-block;
+    width: 0.06rem;
+    margin-top: 0;
+    margin-bottom: 0;
+    border: 0;
+    border-left: 1px solid ${get('semanticColors.border.primary')};
+
+    ${compose(space)}
+`;
+
+const Divider: React.FC<DividerProps> = ({ vertical = false, offset = '1rem' }: DividerProps) =>
```

Comment:
I kept it because we type most Wave components as React.FC, otherwise I wouldn't really put any type here :) There is loads of articles explaining why React.FC should not be really used anymore, e.g. https://fettblog.eu/typescript-react-why-i-dont-use-react-fc/

Reply:
I wouldn't block the merge of this PR because of this, but maybe it's a good moment to stop adding return types to new components.
Check out [this Kent C. Dodds article](https://kentcdodds.com/blog/how-to-write-a-react-component-in-typescript), particularly this bit:

> Ok, so what about the return value? Well, we could type it as React.ReactElement or even wider as a JSX.Element. But honestly, I side with my friend [Nick McCurdy](https://twitter.com/nickemccurdy) when [he says](https://twitter.com/nickemccurdy/status/1365384372908621833) that mistakes can easily be made causing the return type to be too wide. So even outside a react context, I default to not specifying the return type (rely on inference) unless necessary. And that's the case here.

Reply:
Feel free to resolve this if you want to merge and I'll open a new issue to have the discussion there

Reply:
great, I totally agree with you here, I will remove the return type for this PR too!

Reply:
Totally agree with you, I would remove it on this PR, but it is currently enforced by eslint, so let's open a separate issue first as you say, to make the proposal :)